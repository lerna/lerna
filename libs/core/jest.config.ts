/* eslint-disable */
export default {
  displayName: "core",
  preset: "../../jest.preset.js",
  testEnvironment: "node",
  transform: {
    "^.+\\.[tj]sx?$": [
      "ts-jest",
      {
        tsconfig: "<rootDir>/tsconfig.spec.json",
      },
    ],
  },
  // Needed to add "json" to avoid issue resolving spdx-license-ids
  moduleFileExtensions: ["ts", "tsx", "js", "jsx", "json"],
  coverageDirectory: "../../coverage/libs/core",
  testTimeout: 45e3,
};
