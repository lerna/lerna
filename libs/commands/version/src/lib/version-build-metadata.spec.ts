import { promptSelectOne as _promptSelectOne, promptTextInput as _promtTextInput } from "@lerna/core";
import { commandRunner, initFixtureFactory, showCommit } from "@lerna/test-helpers";
import path from "path";

// eslint-disable-next-line jest/no-mocks-import
jest.mock("@lerna/core", () => require("@lerna/test-helpers/__mocks__/@lerna/core"));

const promptTextInput = jest.mocked(_promtTextInput);

// The mocked version isn't the same as the real one
const promptSelectOne = _promptSelectOne as any;

jest.mock("./git-push");
jest.mock("./is-anything-committed", () => ({
  isAnythingCommitted: jest.fn().mockReturnValue(true),
}));
jest.mock("./is-behind-upstream", () => ({
  isBehindUpstream: jest.fn().mockReturnValue(false),
}));
jest.mock("./remote-branch-exists", () => ({
  remoteBranchExists: jest.fn().mockResolvedValue(true),
}));

// eslint-disable-next-line @typescript-eslint/no-var-requires
const { makePromptVersion } = require("./prompt-version");

const resolvePrereleaseId = jest.fn(() => "alpha");
const versionPrompt = (buildMetadata) => makePromptVersion(resolvePrereleaseId, buildMetadata);

// helpers
const initFixture = initFixtureFactory(path.resolve(__dirname, "../../publish/__tests__"));

// file under test
// eslint-disable-next-line @typescript-eslint/no-var-requires
const lernaVersion = commandRunner(require("../command"));

describe("--build-metadata without prompt", () => {
  it("accepts build metadata for explicit version", async () => {
    const testDir = await initFixture("normal");
    await lernaVersion(testDir)("1.0.1", "--build-metadata", "20130313144700");

    expect(promptSelectOne).not.toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("updates build metadata for explicit version", async () => {
    const testDir = await initFixture("build-metadata");
    await lernaVersion(testDir)("1.0.1", "--build-metadata", "20130313144700");

    expect(promptSelectOne).not.toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("accepts build metadata for repository version", async () => {
    const testDir = await initFixture("normal");
    await lernaVersion(testDir)("--repo-version", "1.0.2", "--build-metadata", "21AF26D3--117B344092BD");

    expect(promptSelectOne).not.toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("accepts build metadata with semver keyword", async () => {
    const testDir = await initFixture("normal");
    await lernaVersion(testDir)("minor", "--build-metadata", "001");

    expect(promptSelectOne).not.toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("accepts build metadata with cd version", async () => {
    const testDir = await initFixture("normal");
    await lernaVersion(testDir)("--cd-version", "premajor", "--build-metadata", "exp.sha.5114f85");

    expect(promptSelectOne).not.toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("accepts build metadata with default prerelease id", async () => {
    const testDir = await initFixture("normal");
    await lernaVersion(testDir)("prerelease", "--build-metadata", "20130313144700");

    expect(promptSelectOne).not.toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("accepts build metadata across independent versions with semver keyword", async () => {
    const testDir = await initFixture("independent");
    await lernaVersion(testDir)("minor", "--build-metadata", "001");

    expect(promptSelectOne).not.toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("updates build metadata across independent versions with semver keyword", async () => {
    const testDir = await initFixture("independent-build-metadata");
    await lernaVersion(testDir)("minor", "--build-metadata", "exp.sha.5114f85");

    expect(promptSelectOne).not.toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });
});

describe("--build-metadata with prompt", () => {
  it("accepts build metadata", async () => {
    const testDir = await initFixture("normal");
    promptSelectOne.chooseBump("minor");

    await lernaVersion(testDir)("--build-metadata", "20130313144700");

    expect(promptSelectOne).toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("updates build metadata", async () => {
    const testDir = await initFixture("build-metadata");
    promptSelectOne.chooseBump("minor");

    await lernaVersion(testDir)("--build-metadata", "20130313144700");

    expect(promptSelectOne).toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("accepts build metadata across independent versions", async () => {
    const testDir = await initFixture("independent");
    promptSelectOne.chooseBump("patch");
    promptSelectOne.chooseBump("minor");
    promptSelectOne.chooseBump("major");
    promptSelectOne.chooseBump("minor");
    promptSelectOne.chooseBump("patch");

    await lernaVersion(testDir)("--build-metadata", "21AF26D3--117B344092BD");

    expect(promptSelectOne).toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });

  it("updates build metadata across independent versions", async () => {
    const testDir = await initFixture("independent-build-metadata");
    promptSelectOne.chooseBump("patch");
    promptSelectOne.chooseBump("minor");

    await lernaVersion(testDir)("--build-metadata", "exp.sha.5114f85");

    expect(promptSelectOne).toHaveBeenCalled();

    const patch = await showCommit(testDir);
    expect(patch).toMatchSnapshot();
  });
});

describe("--build-metadata in version prompt", () => {
  test.each([
    ["patch", "1.0.1+001"],
    ["minor", "1.1.0+001"],
    ["major", "2.0.0+001"],
    ["prepatch", "1.0.1-alpha.0+001"],
    ["preminor", "1.1.0-alpha.0+001"],
    ["premajor", "2.0.0-alpha.0+001"],
  ])("accepts build metadata for prompted choice %s", async (bump, result) => {
    promptSelectOne.chooseBump(bump);

    const choice = await versionPrompt("001")({ version: "1.0.0" });

    expect(promptSelectOne).toHaveBeenLastCalledWith(
      "Select a new version (currently 1.0.0)",
      expect.objectContaining({
        choices: expect.any(Array),
      })
    );
    expect(choice).toBe(result);
  });

  it("updates build metadata for prompted choice", async () => {
    promptSelectOne.chooseBump("patch");

    const choice = await versionPrompt("20130313144700")({ version: "1.0.0+001" });

    expect(promptSelectOne).toHaveBeenLastCalledWith(
      "Select a new version (currently 1.0.0+001)",
      expect.objectContaining({
        choices: expect.any(Array),
      })
    );
    expect(choice).toBe("1.0.1+20130313144700");
  });

  it("accepts build metadata for prompted prerelease version", async () => {
    let inputFilter;

    promptSelectOne.chooseBump("PRERELEASE");
    promptTextInput.mockImplementationOnce((msg, cfg) => {
      inputFilter = cfg.filter;
      return Promise.resolve(msg);
    });

    await versionPrompt("exp.sha.5114f85")({ version: "1.0.0" });

    expect(promptSelectOne).toHaveBeenLastCalledWith(
      "Select a new version (currently 1.0.0)",
      expect.objectContaining({
        choices: expect.any(Array),
      })
    );
    expect(inputFilter("rc")).toBe("1.0.1-rc.0+exp.sha.5114f85");
  });

  it("accepts build metadata for prompted custom version", async () => {
    let inputFilter;
    let inputValidate;

    promptSelectOne.chooseBump("CUSTOM");
    promptTextInput.mockImplementationOnce((msg, cfg) => {
      inputFilter = cfg.filter;
      inputValidate = cfg.validate;
      return Promise.resolve(msg);
    });

    await versionPrompt("20130313144700")({ version: "1.0.0" });

    expect(promptSelectOne).toHaveBeenLastCalledWith(
      "Select a new version (currently 1.0.0)",
      expect.objectContaining({
        choices: expect.any(Array),
      })
    );
    expect(inputValidate(inputFilter("2.0.0+20130313144700"))).toBe(true);
  });
});
