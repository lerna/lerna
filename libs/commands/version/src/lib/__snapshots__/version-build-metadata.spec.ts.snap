// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--build-metadata with prompt accepts build metadata 1`] = `
"v1.1.0+20130313144700

HEAD -> main, tag: v1.1.0+20130313144700

diff --git a/lerna.json b/lerna.json
index SHA..SHA 100644
--- a/lerna.json
+++ b/lerna.json
@@ -2 +2 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+20130313144700\\",
diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+20130313144700\\",
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+20130313144700\\",
@@ -5 +5 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.1.0+20130313144700\\"
diff --git a/packages/package-3/package.json b/packages/package-3/package.json
index SHA..SHA 100644
--- a/packages/package-3/package.json
+++ b/packages/package-3/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+20130313144700\\",
@@ -8 +8 @@
-    \\"package-2\\": \\"^1.0.0\\"
+    \\"package-2\\": \\"^1.1.0+20130313144700\\"
diff --git a/packages/package-4/package.json b/packages/package-4/package.json
index SHA..SHA 100644
--- a/packages/package-4/package.json
+++ b/packages/package-4/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+20130313144700\\",
diff --git a/packages/package-5/package.json b/packages/package-5/package.json
index SHA..SHA 100644
--- a/packages/package-5/package.json
+++ b/packages/package-5/package.json
@@ -4 +4 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.1.0+20130313144700\\"
@@ -7 +7 @@
-    \\"package-3\\": \\"^1.0.0\\"
+    \\"package-3\\": \\"^1.1.0+20130313144700\\"
@@ -10 +10 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+20130313144700\\","
`;

exports[`--build-metadata with prompt accepts build metadata across independent versions 1`] = `
"Publish

 - package-1@1.0.1+21AF26D3--117B344092BD
 - package-2@2.1.0+21AF26D3--117B344092BD
 - package-3@4.0.0+21AF26D3--117B344092BD
 - package-4@4.1.0+21AF26D3--117B344092BD
 - package-5@5.0.1+21AF26D3--117B344092BD

HEAD -> main, tag: package-5@5.0.1+21AF26D3--117B344092BD, tag: package-4@4.1.0+21AF26D3--117B344092BD, tag: package-3@4.0.0+21AF26D3--117B344092BD, tag: package-2@2.1.0+21AF26D3--117B344092BD, tag: package-1@1.0.1+21AF26D3--117B344092BD

diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1+21AF26D3--117B344092BD\\",
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"2.0.0\\",
+  \\"version\\": \\"2.1.0+21AF26D3--117B344092BD\\",
@@ -5 +5 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.0.1+21AF26D3--117B344092BD\\"
diff --git a/packages/package-3/package.json b/packages/package-3/package.json
index SHA..SHA 100644
--- a/packages/package-3/package.json
+++ b/packages/package-3/package.json
@@ -3 +3 @@
-  \\"version\\": \\"3.0.0\\",
+  \\"version\\": \\"4.0.0+21AF26D3--117B344092BD\\",
@@ -5 +5 @@
-    \\"package-2\\": \\"^2.0.0\\"
+    \\"package-2\\": \\"^2.1.0+21AF26D3--117B344092BD\\"
diff --git a/packages/package-4/package.json b/packages/package-4/package.json
index SHA..SHA 100644
--- a/packages/package-4/package.json
+++ b/packages/package-4/package.json
@@ -3 +3 @@
-  \\"version\\": \\"4.0.0\\",
+  \\"version\\": \\"4.1.0+21AF26D3--117B344092BD\\",
diff --git a/packages/package-5/package.json b/packages/package-5/package.json
index SHA..SHA 100644
--- a/packages/package-5/package.json
+++ b/packages/package-5/package.json
@@ -4 +4 @@
-    \\"package-3\\": \\"^3.0.0\\"
+    \\"package-3\\": \\"^4.0.0+21AF26D3--117B344092BD\\"
@@ -7 +7 @@
-  \\"version\\": \\"5.0.0\\",
+  \\"version\\": \\"5.0.1+21AF26D3--117B344092BD\\","
`;

exports[`--build-metadata with prompt updates build metadata 1`] = `
"v1.1.0+20130313144700

HEAD -> main, tag: v1.1.0+20130313144700

diff --git a/lerna.json b/lerna.json
index SHA..SHA 100644
--- a/lerna.json
+++ b/lerna.json
@@ -2 +2 @@
-  \\"version\\": \\"1.0.0+001\\"
+  \\"version\\": \\"1.1.0+20130313144700\\"
diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0+001\\"
+  \\"version\\": \\"1.1.0+20130313144700\\""
`;

exports[`--build-metadata with prompt updates build metadata across independent versions 1`] = `
"Publish

 - package-1@1.0.1+exp.sha.SHA
 - package-2@2.1.0+exp.sha.SHA

HEAD -> main, tag: package-2@2.1.0+exp.sha.SHA, tag: package-1@1.0.1+exp.sha.SHA

diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0+001\\"
+  \\"version\\": \\"1.0.1+exp.sha.SHA\\"
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"2.0.0+001\\"
+  \\"version\\": \\"2.1.0+exp.sha.SHA\\""
`;

exports[`--build-metadata without prompt accepts build metadata across independent versions with semver keyword 1`] = `
"Publish

 - package-1@1.1.0+001
 - package-2@2.1.0+001
 - package-3@3.1.0+001
 - package-4@4.1.0+001
 - package-5@5.1.0+001

HEAD -> main, tag: package-5@5.1.0+001, tag: package-4@4.1.0+001, tag: package-3@3.1.0+001, tag: package-2@2.1.0+001, tag: package-1@1.1.0+001

diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+001\\",
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"2.0.0\\",
+  \\"version\\": \\"2.1.0+001\\",
@@ -5 +5 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.1.0+001\\"
diff --git a/packages/package-3/package.json b/packages/package-3/package.json
index SHA..SHA 100644
--- a/packages/package-3/package.json
+++ b/packages/package-3/package.json
@@ -3 +3 @@
-  \\"version\\": \\"3.0.0\\",
+  \\"version\\": \\"3.1.0+001\\",
@@ -5 +5 @@
-    \\"package-2\\": \\"^2.0.0\\"
+    \\"package-2\\": \\"^2.1.0+001\\"
diff --git a/packages/package-4/package.json b/packages/package-4/package.json
index SHA..SHA 100644
--- a/packages/package-4/package.json
+++ b/packages/package-4/package.json
@@ -3 +3 @@
-  \\"version\\": \\"4.0.0\\",
+  \\"version\\": \\"4.1.0+001\\",
diff --git a/packages/package-5/package.json b/packages/package-5/package.json
index SHA..SHA 100644
--- a/packages/package-5/package.json
+++ b/packages/package-5/package.json
@@ -4 +4 @@
-    \\"package-3\\": \\"^3.0.0\\"
+    \\"package-3\\": \\"^3.1.0+001\\"
@@ -7 +7 @@
-  \\"version\\": \\"5.0.0\\",
+  \\"version\\": \\"5.1.0+001\\","
`;

exports[`--build-metadata without prompt accepts build metadata for explicit version 1`] = `
"v1.0.1+20130313144700

HEAD -> main, tag: v1.0.1+20130313144700

diff --git a/lerna.json b/lerna.json
index SHA..SHA 100644
--- a/lerna.json
+++ b/lerna.json
@@ -2 +2 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1+20130313144700\\",
diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1+20130313144700\\",
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1+20130313144700\\",
@@ -5 +5 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.0.1+20130313144700\\"
diff --git a/packages/package-3/package.json b/packages/package-3/package.json
index SHA..SHA 100644
--- a/packages/package-3/package.json
+++ b/packages/package-3/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1+20130313144700\\",
@@ -8 +8 @@
-    \\"package-2\\": \\"^1.0.0\\"
+    \\"package-2\\": \\"^1.0.1+20130313144700\\"
diff --git a/packages/package-4/package.json b/packages/package-4/package.json
index SHA..SHA 100644
--- a/packages/package-4/package.json
+++ b/packages/package-4/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1+20130313144700\\",
diff --git a/packages/package-5/package.json b/packages/package-5/package.json
index SHA..SHA 100644
--- a/packages/package-5/package.json
+++ b/packages/package-5/package.json
@@ -4 +4 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.0.1+20130313144700\\"
@@ -7 +7 @@
-    \\"package-3\\": \\"^1.0.0\\"
+    \\"package-3\\": \\"^1.0.1+20130313144700\\"
@@ -10 +10 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1+20130313144700\\","
`;

exports[`--build-metadata without prompt accepts build metadata for repository version 1`] = `
"v1.0.2+21AF26D3--117B344092BD

HEAD -> main, tag: v1.0.2+21AF26D3--117B344092BD

diff --git a/lerna.json b/lerna.json
index SHA..SHA 100644
--- a/lerna.json
+++ b/lerna.json
@@ -2 +2 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.2+21AF26D3--117B344092BD\\",
diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.2+21AF26D3--117B344092BD\\",
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.2+21AF26D3--117B344092BD\\",
@@ -5 +5 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.0.2+21AF26D3--117B344092BD\\"
diff --git a/packages/package-3/package.json b/packages/package-3/package.json
index SHA..SHA 100644
--- a/packages/package-3/package.json
+++ b/packages/package-3/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.2+21AF26D3--117B344092BD\\",
@@ -8 +8 @@
-    \\"package-2\\": \\"^1.0.0\\"
+    \\"package-2\\": \\"^1.0.2+21AF26D3--117B344092BD\\"
diff --git a/packages/package-4/package.json b/packages/package-4/package.json
index SHA..SHA 100644
--- a/packages/package-4/package.json
+++ b/packages/package-4/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.2+21AF26D3--117B344092BD\\",
diff --git a/packages/package-5/package.json b/packages/package-5/package.json
index SHA..SHA 100644
--- a/packages/package-5/package.json
+++ b/packages/package-5/package.json
@@ -4 +4 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.0.2+21AF26D3--117B344092BD\\"
@@ -7 +7 @@
-    \\"package-3\\": \\"^1.0.0\\"
+    \\"package-3\\": \\"^1.0.2+21AF26D3--117B344092BD\\"
@@ -10 +10 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.2+21AF26D3--117B344092BD\\","
`;

exports[`--build-metadata without prompt accepts build metadata with cd version 1`] = `
"v2.0.0-alpha.0+exp.sha.SHA

HEAD -> main, tag: v2.0.0-alpha.0+exp.sha.SHA

diff --git a/lerna.json b/lerna.json
index SHA..SHA 100644
--- a/lerna.json
+++ b/lerna.json
@@ -2 +2 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"2.0.0-alpha.0+exp.sha.SHA\\",
diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"2.0.0-alpha.0+exp.sha.SHA\\",
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"2.0.0-alpha.0+exp.sha.SHA\\",
@@ -5 +5 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^2.0.0-alpha.0+exp.sha.SHA\\"
diff --git a/packages/package-3/package.json b/packages/package-3/package.json
index SHA..SHA 100644
--- a/packages/package-3/package.json
+++ b/packages/package-3/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"2.0.0-alpha.0+exp.sha.SHA\\",
@@ -8 +8 @@
-    \\"package-2\\": \\"^1.0.0\\"
+    \\"package-2\\": \\"^2.0.0-alpha.0+exp.sha.SHA\\"
diff --git a/packages/package-4/package.json b/packages/package-4/package.json
index SHA..SHA 100644
--- a/packages/package-4/package.json
+++ b/packages/package-4/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"2.0.0-alpha.0+exp.sha.SHA\\",
diff --git a/packages/package-5/package.json b/packages/package-5/package.json
index SHA..SHA 100644
--- a/packages/package-5/package.json
+++ b/packages/package-5/package.json
@@ -4 +4 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^2.0.0-alpha.0+exp.sha.SHA\\"
@@ -7 +7 @@
-    \\"package-3\\": \\"^1.0.0\\"
+    \\"package-3\\": \\"^2.0.0-alpha.0+exp.sha.SHA\\"
@@ -10 +10 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"2.0.0-alpha.0+exp.sha.SHA\\","
`;

exports[`--build-metadata without prompt accepts build metadata with default prerelease id 1`] = `
"v1.0.1-alpha.0+20130313144700

HEAD -> main, tag: v1.0.1-alpha.0+20130313144700

diff --git a/lerna.json b/lerna.json
index SHA..SHA 100644
--- a/lerna.json
+++ b/lerna.json
@@ -2 +2 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1-alpha.0+20130313144700\\",
diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1-alpha.0+20130313144700\\",
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1-alpha.0+20130313144700\\",
@@ -5 +5 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.0.1-alpha.0+20130313144700\\"
diff --git a/packages/package-3/package.json b/packages/package-3/package.json
index SHA..SHA 100644
--- a/packages/package-3/package.json
+++ b/packages/package-3/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1-alpha.0+20130313144700\\",
@@ -8 +8 @@
-    \\"package-2\\": \\"^1.0.0\\"
+    \\"package-2\\": \\"^1.0.1-alpha.0+20130313144700\\"
diff --git a/packages/package-4/package.json b/packages/package-4/package.json
index SHA..SHA 100644
--- a/packages/package-4/package.json
+++ b/packages/package-4/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1-alpha.0+20130313144700\\",
diff --git a/packages/package-5/package.json b/packages/package-5/package.json
index SHA..SHA 100644
--- a/packages/package-5/package.json
+++ b/packages/package-5/package.json
@@ -4 +4 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.0.1-alpha.0+20130313144700\\"
@@ -7 +7 @@
-    \\"package-3\\": \\"^1.0.0\\"
+    \\"package-3\\": \\"^1.0.1-alpha.0+20130313144700\\"
@@ -10 +10 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.0.1-alpha.0+20130313144700\\","
`;

exports[`--build-metadata without prompt accepts build metadata with semver keyword 1`] = `
"v1.1.0+001

HEAD -> main, tag: v1.1.0+001

diff --git a/lerna.json b/lerna.json
index SHA..SHA 100644
--- a/lerna.json
+++ b/lerna.json
@@ -2 +2 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+001\\",
diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+001\\",
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+001\\",
@@ -5 +5 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.1.0+001\\"
diff --git a/packages/package-3/package.json b/packages/package-3/package.json
index SHA..SHA 100644
--- a/packages/package-3/package.json
+++ b/packages/package-3/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+001\\",
@@ -8 +8 @@
-    \\"package-2\\": \\"^1.0.0\\"
+    \\"package-2\\": \\"^1.1.0+001\\"
diff --git a/packages/package-4/package.json b/packages/package-4/package.json
index SHA..SHA 100644
--- a/packages/package-4/package.json
+++ b/packages/package-4/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+001\\",
diff --git a/packages/package-5/package.json b/packages/package-5/package.json
index SHA..SHA 100644
--- a/packages/package-5/package.json
+++ b/packages/package-5/package.json
@@ -4 +4 @@
-    \\"package-1\\": \\"^1.0.0\\"
+    \\"package-1\\": \\"^1.1.0+001\\"
@@ -7 +7 @@
-    \\"package-3\\": \\"^1.0.0\\"
+    \\"package-3\\": \\"^1.1.0+001\\"
@@ -10 +10 @@
-  \\"version\\": \\"1.0.0\\",
+  \\"version\\": \\"1.1.0+001\\","
`;

exports[`--build-metadata without prompt updates build metadata across independent versions with semver keyword 1`] = `
"Publish

 - package-1@1.1.0+exp.sha.SHA
 - package-2@2.1.0+exp.sha.SHA

HEAD -> main, tag: package-2@2.1.0+exp.sha.SHA, tag: package-1@1.1.0+exp.sha.SHA

diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0+001\\"
+  \\"version\\": \\"1.1.0+exp.sha.SHA\\"
diff --git a/packages/package-2/package.json b/packages/package-2/package.json
index SHA..SHA 100644
--- a/packages/package-2/package.json
+++ b/packages/package-2/package.json
@@ -3 +3 @@
-  \\"version\\": \\"2.0.0+001\\"
+  \\"version\\": \\"2.1.0+exp.sha.SHA\\""
`;

exports[`--build-metadata without prompt updates build metadata for explicit version 1`] = `
"v1.0.1+20130313144700

HEAD -> main, tag: v1.0.1+20130313144700

diff --git a/lerna.json b/lerna.json
index SHA..SHA 100644
--- a/lerna.json
+++ b/lerna.json
@@ -2 +2 @@
-  \\"version\\": \\"1.0.0+001\\"
+  \\"version\\": \\"1.0.1+20130313144700\\"
diff --git a/packages/package-1/package.json b/packages/package-1/package.json
index SHA..SHA 100644
--- a/packages/package-1/package.json
+++ b/packages/package-1/package.json
@@ -3 +3 @@
-  \\"version\\": \\"1.0.0+001\\"
+  \\"version\\": \\"1.0.1+20130313144700\\""
`;
